src/Servers/HttpSys/src/RequestProcessing/Request.cs

```diff
@@ -120,7 +120,7 @@ internal Request(RequestContext requestContext)
         internal ulong RawConnectionId { get; }
 
         // No ulongs in public APIs...
-        public long ConnectionId => (long)RawConnectionId;
+        public long ConnectionId => RawConnectionId != 0 ? (long)RawConnectionId : (long)UConnectionId;
```

Comment:
This is not a real suggestion, I just wanted to see how it looks:

```suggestion
        public long ConnectionId => switch RawConnectionId 
        {
           0 => UConnectionId
           _ => RawConnectionId 
        }
```